dts/bindings/wifi/nordic,nrf7002-qspi.yaml

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+
+description: This is a representation of the nRF7002 WiFi chip.
+
+compatible: "nordic,nrf7002-qspi"
+
+include: "nordic,nrf7002.yaml"
+
+on-bus: qspi
+
+properties:
+  sck-frequency:
+    type: int
+    required: true
+    description: |
+      Maximum clock speed supported by the device, in Hz.
+
+  reg:
+    required: true
+
+  label:
+    required: false
+
+  readoc:
+    type: string
+    required: false
+    enum:
+      - "fastread"        # Single data line SPI, FAST_READ (0x0B)
+      - "read2o"          # Dual data line SPI, READ2O (0x3B)
+      - "read2io"         # Dual data line SPI, READ2IO (0xBB)
+      - "read4o"          # Quad data line SPI, READ4O (0x6B)
+      - "read4io"         # Quad data line SPI, READ4IO (0xEB)
+    description: |
+      Specify the number of data lines and opcode used for reading.
+      If not provided fastread will be selected.
+
+  writeoc:
+    type: string
+    required: false
+    enum:
+      - "pp"            # Single data line SPI, PP (0x02)
+      - "pp2o"          # Dual data line SPI, PP2O (0xA2)
+      - "pp4o"          # Quad data line SPI, PP4O (0x32)
+      - "pp4io"         # Quad data line SPI, PP4IO (0x38)
+    description: |
+      Specify the number of data lines and opcode used for writing.
+      If not provided pp will be selected.
```

Comment:
@peknis  @greg-fer  @OisinFitz  :  whenever you decide who is going to own the documentation for this PR, please recall that devicetree bindings descriptions are also user-facing documentation and should be held to the same standards as anything that appears in the .rst.

All of our company's devicetree bindings documentation is autogenerated from these files and appears here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/build/dts/api/bindings.html#dt-vendor-nordic

Please review these descriptions too :). Thanks.

Reply:
Closing this now, as I got comments from @peknis on dts bindings.